### PR TITLE
docs: Add a note on supported backends

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

**Add a one line overview of what this PR aims to accomplish.**

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA/NeMo-RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA/NeMo-RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA/NeMo-RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,6 +309,47 @@ sbatch \
     ray.sub
 ```
 
+## Supported Backends
+
+NeMo RL supports multiple training backends to accommodate different model sizes and hardware configurations:
+
+### Available Backends
+
+- **FSDP1** - Fully Sharded Data Parallel (legacy implementation)
```
Comment (terrykong) [inline README.md]: let's not even mention FSDP1 since it's on the way out

File: README.md [hunk 2]
```diff
@@ -308,6 +309,47 @@ sbatch \
...
+- **DTensor (FSDP2)** - PyTorch's next-generation distributed training with improved memory efficiency
+- **Megatron** - NVIDIA's high-performance training framework for scaling to large models (>100B parameters)
+
+### Backend Selection
```
Comment (parthchadha) [inline README.md]: This section seems too much text for README? I would personally prefer adding the section with key highlights and then linking to our docs for further detailed information about how to switch backends etc.

File: docs/design-docs/supported-backends.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Supported Backends
```
Comment (terrykong) [inline docs/design-docs/supported-backends.md]: can we title this `Training Backends`?

File: README.md (hunk 2 above)
Comment (terrykong) [reply README.md]: i feel like we should maybe mention a run command (w/o a lot of verbiage) under single gpu instructions to show how easy it is to try mcore out. wdyt @parthchadha ?

Comment (ashors1) [reply README.md]: @terrykong this is okay with me, but just to let you know, this would also require adding a `megatron_cfg` section to the `grpo_math_1B.yaml` config to make it easy to switch between megatron and dtensor via the command line, which would make that config quit a bit more verbose.

Alternatively, we could add an example run command that uses `grpo_math_1B_megatron.yaml` directly, but that would hide how the config needs to be modified to switch between dtensor and megatron.

Comment (parthchadha) [reply README.md]: Sure, lets go with `grpo_math_1B_megatron.yaml` and point to docs for better understanding of the megatron_cfg yaml structure.

File: docs/design-docs/training-backends.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+- **Megatron** - NVIDIA's high-performance training framework for scaling to large models (>100B parameters)
+
+## Backend Selection
+
+The training backend is automatically determined based on your YAML configuration settings. Here's how to configure each backend.
+
+### Megatron Backend
+To enable Megatron-based training:
+
+1. Add the `megatron_cfg` key to your policy configuration
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
1. Add the `megatron_cfg` key to your policy configuration.
```

File: docs/design-docs/training-backends.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+2. Set `policy.megatron_cfg.enabled=True`
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
2. Set `policy.megatron_cfg.enabled=True`.
```

File: docs/design-docs/training-backends.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+3. Refer to [examples/configs/grpo_math_1B_megatron.yaml](../../examples/configs/grpo_math_1B_megatron.yaml) for a complete configuration example
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
3. Refer to [examples/configs/grpo_math_1B_megatron.yaml](../../examples/configs/grpo_math_1B_megatron.yaml) for a complete configuration example.
```

File: docs/design-docs/training-backends.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+_Note_: When using Megatron, the optimizer and learning rate schedule are configured through `policy.megatron_cfg.optimizer` and `policy.megatron_cfg.scheduler`, respectively.
+
+### DTensor Backend
+To enable DTensor (FSDP2) training:
+
+1. Set `policy.dtensor_config.enabled=True`
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
1. Set `policy.dtensor_config.enabled=True`.
```

File: docs/design-docs/training-backends.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+2. Refer to [examples/configs/grpo_math_1B.yaml](../../examples/configs/grpo_math_1B.yaml) for a configuration example
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
2. Refer to [examples/configs/grpo_math_1B.yaml](../../examples/configs/grpo_math_1B.yaml) for a configuration example.
```

File: docs/design-docs/training-backends.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+## Backend Priority
+
+**Megatron takes precedence over DTensor.** If both backends are enabled simultaneously (`policy.megatron_cfg.enabled=True` and `policy.dtensor_config.enabled=True`), the Megatron backend will be used.
+
+## Configuration Examples
+
+For comprehensive examples of each algorithm with each backend, see the [examples/configs/recipes/llm](../../examples/configs/recipes/llm/) folder. This directory contains ready-to-use configurations for various supported combinations.
```
Comment (jgerh) [inline docs/design-docs/training-backends.md]:
```suggestion
For comprehensive examples of each algorithm and backend, see the [examples/configs/recipes/llm](../../examples/configs/recipes/llm/) folder. This directory contains ready-to-use configurations for various supported combinations.
```

File: README.md [hunk 10]
```diff
@@ -150,6 +151,16 @@ uv run python examples/run_grpo_math.py \
   logger.num_val_samples_to_print=10
 ```
 
+The default configuration uses the DTensor training backend. We also provide a config `examples/configs/grpo_math_1B_megatron.yaml` which is set up to use the Megatron backend out of the box. To train using this config on a single GPU,
```
Comment (jgerh) [inline README.md]:
```suggestion
The default configuration uses the DTensor training backend. We also provide a config `examples/configs/grpo_math_1B_megatron.yaml` which is set up to use the Megatron backend out of the box. 

To train using this config on a single GPU:
```

**Review by terrykong — APPROVED** [review]
